Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs

```diff
@@ -988,3 +995,107 @@ updateByPhoneNumber merchantShortId phoneNumber req = do
       AV.create aadhaarEntity
   QDriverInfo.updateAadhaarVerifiedState (cast driver.id) True
   pure Success
+
+fleetUnlinkVehicle :: ShortId DM.Merchant -> Text -> Text -> Text -> Flow APISuccess
+fleetUnlinkVehicle merchantShortId fleetOwnerId vehicleNo newDriverMobileNo = do
+  merchant <- findMerchantByShortId merchantShortId
+  vehicle <- QVehicle.findByRegistrationNo vehicleNo >>= fromMaybeM (VehicleDoesNotExist vehicleNo)
+  unless (isJust vehicle.fleetOwnerId && vehicle.fleetOwnerId == Just fleetOwnerId) $ throwError (FleetOwnerVehicleMismatchError fleetOwnerId)
+  let oldDriverId = vehicle.driverId
+  mobileNumberHash <- getDbHash newDriverMobileNo
+  newDriver <- QPerson.findByMobileNumberAndMerchant mobileIndianCode mobileNumberHash merchant.id >>= fromMaybeM (PersonDoesNotExist newDriverMobileNo)
+  unless (oldDriverId /= newDriver.id) $ throwError (DriverAlreadyLinkedWithVehicle $ show newDriver.id)
+  DomainRC.deactivateCurrentRC oldDriverId
+  _ <- QVehicle.deleteById oldDriverId
+  _ <- QDriverInfo.updateEnabledVerifiedState oldDriverId False False
+  logTagInfo "dashboard -> unlinkVehicle : " (show oldDriverId)
+  Redis.set (DomainRC.makeFleetOwnerKey newDriver.id.getId) fleetOwnerId -- setting this value here , so while creation of creation of vehicle we can add fleet owner id
```

Comment:
Here This should not be driver ID it should be vehicle Number.

Reply:
Changed